lld/MachO/SyntheticSections.cpp

```diff
@@ -1251,26 +1255,26 @@ void SymtabSection::emitStabs() {
 
       // We use 'originalIsec' to get the file id of the symbol since 'isec()'
       // might point to the merged ICF symbol's file
-      symbolsNeedingStabs.emplace_back(defined,
-                                       defined->originalIsec->getFile()->id);
+      Defined *funcBodySym = getFuncBodySym(defined);
```

Comment:
Do you plan to use `getFuncBodySym()` elsewhere? Because I think it'll be more readable to inline it here.
```suggestion
      Defined *funcBodySym = defined->identicalCodeFoldingKind == Symbol::ICFFoldKind::Thunk ? macho::getBodyForThunkFoldedSym(defined) : defined;
```

Reply:
The logic seems pretty clear IMO: "If this is a thunk, call this special thunk function".

If you want to keep the function, then maybe we should rename it and add some docs. In the most common case, when `defined` is not a thunk, it is very mysterious why we need to pass it through `getFuncBodySym()` first.